### PR TITLE
Refactor the command line option handling of the `jerry` tool

### PR DESCRIPTION
**Pull request by akosthekiss**
The file `main-unix.c` contained a lot of recurring patterns in the
command line option handling parts. This patch removes these code
clones with the introduction of two helper macros. The goal of the
patch is to have a smaller `jerry` tool, both source and binary
wise.

JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

### CODE REVIEW

**Comment by akosthekiss** [comment]
Note: the first version of this patch had static functions for `check_feature` and `check_usage`. Interestingly, they have _increased_ the binary. Interestingly no.2, the use of macros has _decreased_ the size of the tool.

File: jerry-main/main-unix.c [hunk 1]
```diff
@@ -423,182 +458,93 @@ main (int argc,
...
-      {
-        jerry_port_default_set_log_level (JERRY_LOG_LEVEL_WARNING);
-        jerry_port_log (JERRY_LOG_LEVEL_WARNING,
-                        "Ignoring 'show-opcodes' option because this feature is disabled!\n");
-      }
     }
     else if (!strcmp ("--show-regexp-opcodes", argv[i]))
     {
-      if (jerry_is_feature_enabled (JERRY_FEATURE_PARSER_DUMP))
+      if (CHECK_FEATURE (JERRY_FEATURE_PARSER_DUMP, argv[i]))
```
Comment (akosthekiss) [inline jerry-main/main-unix.c]: Question to self/anyone: was this OK before the patch? Shouldn't this line check for `JERRY_FEATURE_REGEXP_DUMP`?

File: jerry-main/main-unix.c [hunk 2]
```diff
@@ -375,6 +375,41 @@ register_js_function (const char *name_p, /**< name of the function */
...
+
+/**
+ * Check whether a usage-related condition holds. If not, print an error
+ * message, print the usage, and terminate the application.
+ *
+ * @param CONDITION the condition that must hold
+ * @param NAME name of the application (argv[0])
+ * @param ... error message to print if condition does not hold
+ */
+#define CHECK_USAGE(CONDITION, NAME, ...) \
```
Comment (zherczeg) [inline jerry-main/main-unix.c]: This could be a function. Too ugly this way.

File: jerry-main/main-unix.c [hunk 3]
```diff
@@ -375,6 +375,41 @@ register_js_function (const char *name_p, /**< name of the function */
...
+ * @param FEATURE feature to check
+ * @param OPTION command line option that triggered this check
+ * @return the status of the feature.
+ */
+#define CHECK_FEATURE(FEATURE, OPTION) \
+  jerry_is_feature_enabled (FEATURE) \
+  ? true \
+  : (jerry_port_default_set_log_level (JERRY_LOG_LEVEL_WARNING), \
+     jerry_port_log (JERRY_LOG_LEVEL_WARNING, "Ignoring '%s' option because this feature is disabled!\n", (OPTION)), \
+     false)
```
Comment (LaszloLango) [inline jerry-main/main-unix.c]: unusual style. Please put every parameter to a separate line and indent properly.

**Comment by akosthekiss** [comment]
I've updated the PR according to the comments

File: jerry-main/main-unix.c [hunk 4]
```diff
@@ -375,6 +375,49 @@ register_js_function (const char *name_p, /**< name of the function */
...
 } /* register_js_function */
 
+/**
+ * Check whether JerryScript has a requested feature enabled or not. If not,
+ * print a warning message.
+ *
+ * @return the status of the feature.
+ */
+static inline bool
+check_feature (jerry_feature_t feature, /**< feature to check */
```
Comment (zherczeg) [inline jerry-main/main-unix.c]: I think this function is not performance critical to be inline.

**Comment by akosthekiss** [comment]
I've updated the PR (removed the `inline`s)

**Pull request merged**